langchain/src/prompts/chat.ts

```diff
@@ -0,0 +1,140 @@
+import { ChatMessage, Role } from "../chat_models/base.js";
+import {
+  BasePromptTemplate,
+  BasePromptTemplateInput,
+  InputValues,
+  PartialValues,
+} from "./base.js";
+import { DEFAULT_FORMATTER_MAPPING, TemplateFormat } from "./template.js";
+import { SerializedPromptTemplate } from "./prompt.js";
+
+export type PromptMessage = {
+  role: Role;
+  message: BasePromptTemplate;
+};
+
+export interface ChatPromptTemplateInput extends BasePromptTemplateInput {
+  /**
+   * The prompt messages
+   */
+  promptMessages: PromptMessage[];
+
+  /**
+   * The format of the prompt template. Options are 'f-string', 'jinja-2'
+   *
+   * @defaultValue 'f-string'
+   */
+  templateFormat?: TemplateFormat;
+
+  /**
+   * Whether or not to try validating the template on initialization
+   *
+   * @defaultValue `true`
+   */
+  validateTemplate?: boolean;
+}
+
+export class ChatPromptTemplate
+  extends BasePromptTemplate
+  implements ChatPromptTemplateInput
+{
+  promptMessages: PromptMessage[];
+
+  templateFormat: TemplateFormat = "f-string";
+
+  validateTemplate = true;
+
+  constructor(input: ChatPromptTemplateInput) {
+    super(input);
+    Object.assign(this, input);
+
+    if (this.validateTemplate) {
+      if (!(this.templateFormat in DEFAULT_FORMATTER_MAPPING)) {
+        const validFormats = Object.keys(DEFAULT_FORMATTER_MAPPING);
+        throw new Error(`Invalid template format. Got \`${this.templateFormat}\`;
+                         should be one of ${validFormats}`);
+      }
+      // create a set of all input variables from all prompt messages
+      const inputVariables = new Set<string>();
```

Comment:
I'm still not sold on the value of specifying this in addition to the prompts, what's the use case that is unlocked by this extra chance for the user making an error?

Reply:
yeah fair enough, will change

Reply:
This is an artifact of `BasePromptTemplate`, and we'll need to change it there. This is outside the scope of this PR but we should definitely address. I don't think input variables should be specified anywhere, they should be just be inferred from the template.